terminate called after throwing an instance of 'rational::zero_denominator'
  what():  denominator is zero
